aosvc-python/aosvc

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+import argparse
+import configparser
+import http.server
+import json
+import logging
+import pathlib
+
+
+logging.basicConfig(format='%(asctime)s %(levelname)s %(message)s', level=logging.INFO)
+logger = logging.getLogger()
+
+
+class MyRequestHandler(http.server.BaseHTTPRequestHandler):
+    def _resp(self, status_code=http.HTTPStatus.OK, content=None):
+        self.send_response(status_code)
+        self.send_header('Access-Control-Allow-Origin', '*')
+        self.send_header('Access-Control-Allow-Methods', 'POST, OPTIONS')
+        self.send_header('Access-Control-Allow-Private-Network', 'true')
+        self.send_header('Access-Control-Allow-Headers', 'Content-Type, Content-Length, Accept-Encoding, accept, origin, Access-Control-Allow-Private-Network')
+        self.send_header('Connection', 'close')
+        if content is not None:
+            if hasattr(content, 'encode'):
+                content = content.encode()
+            elif not isinstance(content, str):
+                content = str(content).encode()
+            self.send_header('Content-Length', str(len(content)))
+        self.end_headers()
+
+        if content is not None:
+            self.wfile.write(content)
+
+    def parse_request(self):
+        if not super().parse_request():
+            logger.error(f'bad request! {self.raw_requestline!r}')
+            return False
+
+        if self.path.rstrip('/') != '/update':
+            self._resp(http.HTTPStatus.NOT_FOUND)
+            return False
+
+        return True
+
+    def do_OPTIONS(self):
+        return self._resp(http.HTTPStatus.NO_CONTENT)
+
+    def do_POST(self):
+        try:
+            credentials = json.loads(self.rfile.read(int(self.headers['Content-Length'])))
+        except Exception as e:
+            logger.exception('Error parsing request!')
+            return self._resp(http.HTTPStatus.BAD_REQUEST, f'{type(e).__name__}: {e}')
+
+        try:
+            aws_credentials_path = pathlib.Path(f'~/.aws/credentials').expanduser()
+            ini = configparser.ConfigParser()
+            ini.read(aws_credentials_path)
+            section = ini['default']
```

Comment:
It seems this fails if the user either doesn't have an explicit `[default]` section created and/or `.aws/credentials` doesn't exist
```
2023-04-20 12:44:28,068 ERROR Error updating credentials file!
Traceback (most recent call last):
  File "/Users/enrique/repos/aws_alwayson/aosvc-python/aosvc", line 58, in do_POST
    section = ini['default']
              ~~~^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/configparser.py", line 979, in __getitem__
    raise KeyError(key)
KeyError: 'default'
```
Can you add the logic for adding those?

Reply:
Fixed. Thanks for testing my code :)